components/builder-jobsrv/src/server/log_archiver/s3.rs

```diff
@@ -158,30 +106,17 @@ impl LogArchiver for S3Archiver {
         request.bucket = self.bucket.clone();
         request.key = Self::key(job_id);
 
-        // As above when uploading a job file, we currently need to
-        // catch a potential panic if the object store cannot be reached
-        let result =
-            panic::catch_unwind(AssertUnwindSafe(|| self.client.get_object(&request, None)));
-
-        let body = match result {
-            Ok(Ok(response)) => response.body, // normal result
-            Ok(Err(e)) => {
-                // This is a "normal", non-panicking error, e.g.,
-                // they're configured with a non-existent bucket.
-                return Err(Error::JobLogRetrieval(job_id, e));
-            }
+        let payload = self.client.get_object(&request).sync();
+        let stream = match payload {
+            Ok(response) => response.body.expect("Downloaded object is not empty"),
```

Comment:
Does `.expect()`  cause a panic when it fires ?Also would the issue here be that the body in the response _is_ empty rather than _not_ empty?

Reply:
Yeah it will panic - I followed basically the pattern from your builder-depot S3 backend here.  GetObjectOutput should return an optional StreamingBody, which we want to be non-empty.